autoload/neomake/makers/mvn.vim

```diff
@@ -4,6 +4,29 @@ function! neomake#makers#mvn#mvn()
     return {
          \ 'exe': 'mvn',
          \ 'args': ['install'],
-            \ 'errorformat': '[%tRROR]\ %f:[%l]\ %m,%-G%.%#'
+         \ 'remove_invalid_entries' : 1,
+         \ 'postprocess': function('neomake#makers#mvn#InvalidDuplicates'),
+         \ 'errorformat':
+           \ '%E[%tRROR]\ %f:[%l]\ %m,' .
+           \ '%E[%tRROR]\ %f:[%l\,%v]\ %m,' .
+           \ '%C\ %s:\ %m,' .
+           \ '%C[ERROR]\ %s:\ %m,' .
+           \ '%-G%.%#'
          \ }
 endfunction
+
+function! neomake#makers#mvn#InvalidDuplicates(entry) abort
+    let entry = a:entry
+    let text = entry.text
+    let job = neomake#GetJobs()[entry.job_id]
+
+    if !has_key(job, 'lines')
+        let job.lines = []
+    endif
+
+    if index(job.lines, text) == -1
+        call add(job.lines, text)
+    else
+        let entry.valid = 0
+    endif
```

Comment:
You should not modify `job.lines` here, but instead just keep track of the seen lines yourself.
The postprocessing is meant to massage the location/quickfix list only.

See https://github.com/neomake/neomake/issues/766, where I thought about passing `jobinfo` into the `postprocess` function, which you could use then - and we would not have to change the `entry` temporarily.